engines/engine.cpp

```diff
@@ -465,30 +465,42 @@ void GUIErrorMessageFormat(Common::U32String fmt, ...) {
  *
  * @return			true if audio files of the expected naming scheme are found, as long as ScummVM
  *					is also built with support to the respective audio format (eg. ogg, flac, mad/mp3)
+ * @note			Make sure to keep this in sync with DefaultAudioCDManager::play()
  */
 bool Engine::existExtractedCDAudioFiles() {
+	const char *extensions[] = {
 #ifdef USE_VORBIS
-	if (Common::File::exists("track1.ogg") ||
-	    Common::File::exists("track01.ogg"))
-		return true;
+		"ogg",
 #endif
 #ifdef USE_FLAC
-	if (Common::File::exists("track1.fla")  ||
-	    Common::File::exists("track1.flac") ||
-	    Common::File::exists("track01.fla") ||
-	    Common::File::exists("track01.flac"))
-		return true;
+		"fla", "flac",
 #endif
 #ifdef USE_MAD
-	if (Common::File::exists("track1.mp3") ||
-	    Common::File::exists("track01.mp3"))
-		return true;
+		"mp3",
 #endif
+		nullptr
+	};
+
+	const char *trackName[] = {
+		"track1",
+		"track01",
+		"track_1",
+		"track_01"
+	};
+
+	for (int i = 0; i < ARRAYSIZE(trackName); ++i) {
+		for (const char **ext = extensions; *ext; ++ext) {
+			const Common::String &filename = Common::String::format("%s.%s", trackName[i], *ext);
+			if (Common::File::exists(filename)) {
+				return true;
+			}
+		}
+	}
```

Comment:
It would be preferable to use the same code in both places, if possible

Reply:
Tried to do that in a follow up commit. The problem is that the supported file format extensions are still duplicated. We need a way to query those extensions. I've seen them in a few other locations spread over the code. Would be easier (as in less `#ifdef USE_THISANDTHAT` in the engine code) for the engine coders.

There is e.g. `SoundHE::tryLoadSoundOverride` - and it looks like it is missing `m4a` support - that might be on purpose though... not sure.